### DISCUSSION TITLE
blueprint triggers

### DISCUSSION THREAD

**Discussion by rananna** [greghesp/ha-bambulab#266] [General]
I can't get the blueprint to trigger o0n the printer current stage.
the sensor is reporting the correct stage,, but the blueprint will not trigger the wled .
For example on completion to change to green.
any ideas?

**Comment by rananna** [comment]
Is the state check in the blueprint conditionals case sensitive?
ie: should state: Cleaning nozzle tip actually be state: cleaning nozzle tip?
I will modify the blueprint to see if that helps with the trigger issue I am having

**Comment by rananna** [comment]
one problem, on completion of print the stage reports unknown, not finish.
developer tools reports current stage options as:
options: offline, unknown, printing, auto_bed_leveling, heatbed_preheating, sweeping_xy_mech_mode, changing_filament, m400_pause, paused_filament_runout, heating_hotend, calibrating_extrusion, scanning_bed_surface, inspecting_first_layer, identifying_build_plate_type, calibrating_micro_lidar, homing_toolhead, cleaning_nozzle_tip, checking_extruder_temperature, paused_user, paused_front_cover_falling, calibrating_extrusion_flow, paused_nozzle_temperature_malfunction, paused_heat_bed_temperature_malfunction, idle

Should the blueprint reference these thye same? ie: state: heatbed_preheating or state: heatbed preheating

**Comment by rananna** [comment]
Well, I got it working with a modified blueprint.
here it is:

blueprint:
  name: Bambu Lab WLED Controller
  description: Control one or more WLED lights with Bambu 3d printer
  domain: automation
  input:
    printer_status:
      name: Print Status of your Bambu Lab printer
      description: Select the print status entity.
      selector:
        entity:
          filter:
          - domain:
            - sensor
            integration: bambu_lab
          multiple: false
    printer_stage:
      name: Current Stage of your Bambu Lab Printer
      description: Select the printer current stage entity.
      selector:
        entity:
          filter:
          - domain:
            - sensor
            integration: bambu_lab
          multiple: false
    chamber_light:
      name: Chamber Light of your Bambu Lab Printer
      description: Select the printer chamber light entity.
      selector:
        entity:
          filter:
          - domain:
            - light
            integration: bambu_lab
          multiple: false
    wled_lights:
      name: WLED Light devices
      description: Select the WLED light devices you wish to control.
      selector:
        device:
          filter:
          - integration: wled
          multiple: true
  source_url: https://github.com/greghesp/ha-bambulab/blob/main/blueprints/wled_controller.yaml
alias: Bambu Lab WLED Controller
description: ''
trigger:
- platform: state
  entity_id:
  - !input chamber_light
  - !input printer_status
  - !input printer_stage
  alias: When printer stage,status or chamber light state changes
condition: []
action:
- if:
  - condition: state
    state: 'on'
    alias: chamber light is on
    entity_id: !input chamber_light
  - condition: not
    conditions:
    - condition: state
      entity_id: !input printer_stage
      state: scanning_bed_surface
      alias: Scanning bed surface
    - condition: state
      entity_id: !input printer_stage
      state: cleaning_nozzle_tip
      alias: Cleaning nozzle Tip
    - condition: state
      entity_id: !input printer_stage
      state: Calibrating_extrusion
      alias: Calibrating extrusion
    - condition: state
      entity_id: !input printer_stage
      state: calibrating_micro_lidar
      alias: Calibrating micro lidar
    - condition: state
      entity_id: !input printer_stage
      state: calibrating_extrusion_flow
      alias: Calibrating extrusion flow
    - condition: state
      entity_id: !input printer_status
      state: offline
      alias: Printer is off
    alias: And Lidar is NOT on
  then:
  - service: light.turn_on
    data:
      rgb_color:
      - 255
      - 255
      - 255
      brightness_pct: 100
      effect: Solid
    alias: Turn WLED light on white
    target:
      device_id: !input wled_lights
  - choose:
    - conditions:
      - condition: or
        conditions:
        - condition: state
          entity_id: !input printer_stage
          state: paused_filament_runout
        - condition: state
          entity_id: !input printer_stage
          state: paused_front_cover_falling
        - condition: state
          entity_id: !input printer_stage
          state: paused_nozzle_temperature_malfunction
        - condition: state
          entity_id: !input printer_stage
          state: paused_heat_bed_temperature_malfunction
        - condition: state
          entity_id: !input printer_stage
          state: paused_user
      sequence:
      - service: light.turn_on
        data:
          rgb_color:
          - 255
          - 0
          - 0
          brightness: 255
          effect: Blink
        target:
          device_id: !input wled_lights
        alias: Turn on Wled to red
    alias: Do we need to set color to red?
  - choose:
    - conditions:
      - condition: or
        conditions:
        - condition: state
          entity_id: !input printer_stage
          state: idle
        - condition: state
          entity_id: !input printer_status
          state: finish
        - condition: state
          entity_id: !input printer_status
          state: unknown
      sequence:
      - service: light.turn_on
        data:
          rgb_color:
          - 0
          - 255
          - 0
          brightness: 255
          effect: Solid
        target:
          device_id: !input wled_lights
      
        alias: Turn on Wled to green
    alias: Do we need to set color to green?
  - choose:
    - conditions:
      - condition: or
        conditions:
        - condition: state
          entity_id: !input printer_stage
          state: auto_bed_leveling
      sequence:
      - service: light.turn_on
        data:
          rgb_color:
          - 0
          - 0
          - 255
          brightness: 255
          effect: Blink
        target:
          device_id: !input wled_lights
      
        alias: Turn on Wled to blue
    alias: Do we need to set color to blue?
  - choose:
    - conditions:
      - condition: or
        conditions:
        - condition: state
          entity_id: !input printer_stage
          state: heatbed_preheating
        - condition: state
          entity_id: !input printer_stage
          state: heating_hotend
      sequence:
      - service: light.turn_on
        data:
          rgb_color:
          - 255
          - 169
          - 0
          brightness: 255
          effect: Blink
        target:
          device_id: !input wled_lights
         
        alias: Turn on Wled to yellow
    alias: Do we need to set color to yellow?
  else:
  - service: light.turn_off
    data: {}
    target:
      device_id: !input wled_lights
     
    alias: Turn WLED light off
mode: single

**Comment by AdrianGarside** [comment]
This should have been fixed with: 
https://github.com/greghesp/ha-bambulab/pull/258

Which looks to be the same change. I'm going to close this discussion but please let me know if I'm wrong.
